### PR TITLE
Fix: Java/Android release

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

- Changed android release building to ubuntu
- Removed -new prefix
- removed building from rocksdb-v18 branch
- cleaned up unused parts of action
- updates java bindings top rc-3
- upgrade gradle wrapper to 8.1

Addresses #177

### CODE REVIEW

File: .github/workflows/java_binding_publish.yml [hunk 1]
```diff
@@ -153,43 +154,24 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-latest]
+        os: [ubuntu-latest]
 
     steps:
       - uses: actions/checkout@v3
-        with:
-          ref: rocksdb-v18
```
Comment (thibault-martinez) [inline .github/workflows/java_binding_publish.yml:161]: We sure about that?

Comment (kwek20) [reply .github/workflows/java_binding_publish.yml:161]: I ran it. Builds fine!

Comment (kwek20) [reply .github/workflows/java_binding_publish.yml:161]: Besides. We deleted that branch haha

File: .github/workflows/java_binding_publish.yml
Comment (Thoralf-M) [inline .github/workflows/java_binding_publish.yml:1]: Probably good to have client or wallet in the file name (also for the other file)

File: .github/workflows/java_binding_publish.yml (hunk 1 above)
Comment (thibault-martinez) [reply .github/workflows/java_binding_publish.yml:161]: We didn't delete it, we just didn't import it from wallet.rs, but if it's good no worries

File: .github/workflows/java_binding_publish.yml
Comment (thibault-martinez) [reply .github/workflows/java_binding_publish.yml:1]: I think AlexC is doing it in his PR, let's not conflict even more

File: sdk/src/wallet/bindings/java/build.gradle [hunk 2]
```diff
@@ -3,7 +3,7 @@ plugins {
 }
 
 group 'org.iota'
-version = '1.0.0-rc.1'
+version = '1.0.0-rc.3'
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/java/build.gradle:6]: Was RC2 ever released? Why the jump? If it was never released, can you make it RC2 and edit the changelog?

Comment (kwek20) [reply sdk/src/wallet/bindings/java/build.gradle:6]: no never, but i figured we align them anyways as we will merge them. ID rather not spend more time on this, as it has 0 uses on maven central anyways

Comment (thibault-martinez) [reply sdk/src/wallet/bindings/java/build.gradle:6]: Yeah sure, just weird since you are going to release it, but whatever

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**